clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1797,4 +1797,18 @@ private void release() {
         if (refcount.decrementAndGet() == 0)
             currentThread.set(NO_CURRENT_THREAD);
     }
+    
+     /**
+      * Returns a flattened Properties object
+      * @param properties - Java Properties object
+      */
+     private static Properties getFlattenedProperties(Properties properties) {
```

Comment:
To avoid the duplication of this method, could we add a new constructor to `AbstractConfig` which accepts a `Properties` object and let it implement `getFlattenedProperties`?